blackbox/docs/administration/ssl.txt

```diff
@@ -0,0 +1,116 @@
+should be signed by a certificate authority (CA). A third-party CA or your
+organization's existing CA can be used. When a client connects to a node using
+SSL/TLS the client receives the certificate provided by the node and will
+determine if the node’s certificate is valid, trusted, and matches the hostname
+or IP address it is trying to connect to.
+
+.. NOTE::
+
+    Technically it's possible to disable CA checks for certificates on the
+    client. It's strongly recommended though to use certificates signed by an
+    official CA or by a private CA (company PKI) that is also known to the
+    client to be able to establish trust between nodes and clients.
+
+Once the keystore is prepared define the absolute file path of the keystore
+``.jks`` file on the node using ``ssl.keystore_filepath`` setting.
+
+.. NOTE::
+
+    Make sure that the keystore file has the right permissions and is accessible
+    by system user ``crate``.
+
+Also define the password needed to decrypt the keystore by using the
+``ssl.keystore_password`` setting.
+
+Use ``ssl.keystore_key_password`` setting to define the key password used
+when creating the keystore.
+
+A full list of the settings needed to configure SSL/TLS can
+be found :ref:`here <ssl_config>`.
+
+Configure a separate TrustStore
+-------------------------------
+
+Trusted CA certificates can also be stored in a node's keystore but optionally
+a separate truststore can be used to store them.
+
+If you want to use a separate truststore, create a node truststore and import
+the CA certificate(s) you want to trust. Once the truststore is prepared
+define the absolute file path of the truststore ``.jks`` file on the node
+using the ``ssl.truststore_filepath`` setting.
+
+.. NOTE::
+
+    Make sure that the truststore file has the right permissions and is
+    accessible by system user ``crate``.
+
+Also define the password needed to decrypt the keystore by using the
+``ssl.truststore_password`` setting.
+
+A full list of the settings needed to configure SSL/TLS can
+be found :ref:`here <ssl_config>`.
+
+
+Connecting to a CrateDB node using Postgres Wire Protocol with SSL/TLS
+----------------------------------------------------------------------
+
+Connect to a CrateDB node using JDBC
+....................................
+
+JDBC needs to validate the CrateDB node's identity by checking that the node
+certificate is signed by a trusted authority. If the certificate is signed by
+a certificate authority (CA) that is known to the java runtime, there is
+nothing further to do (Java comes with copies of the most common CA's certificates).
+
+If you have a certificate that is signed by a CA not known to the java runtime,
+you need to configure a truststore which contains the node's certificate and
+provide the path to the truststore file along with the password when starting
+your java application::
+
+    java -Djavax.net.ssl.trustStore=mystore -Djavax.net.ssl.trustStorePassword=mypassword com.mycompany.MyApp
```

Comment:
It is possible to use the `sslrootcert` connection parameter:

https://jdbc.postgresql.org/documentation/head/connect.html